api/pkg/cli/cmd/install/install.go

```diff
@@ -134,13 +134,21 @@ func (opts *options) run() error {
 		}
 	}
 
-	installer := installer.New(opts.cs)
-	opts.resource, err = installer.Install(manifest, opts.from, opts.cs.Namespace())
-	if err != nil {
-		return opts.errors(err)
+	out := opts.cli.Stream().Out
+
+	resourceInstaller := installer.New(opts.cs)
+
+	resource, errors := resourceInstaller.Install(manifest, opts.from, opts.cs.Namespace())
+	opts.resource = resource
+
+	if len(errors) != 0 {
+		// process the errors and return the response
+		err = opts.errors(resourceInstaller.PipelineVersion, errors)
+		if err != nil {
```

Comment:
we are checking `len(error) != 0` and then processing the error, and again checking `err != nil` 😄 ? is it required this way

Reply:
we can do 
```
if err[0] == ErrWarnVersionNotFound && len(err) == 1{
     WARN
}else{
      return opt.errors()
}
```